File: javascript/kindergarten-garden/_iterations/1/kindergarten-garden.js
const DEFAULT_STUDENTS = ['alice', 'bob', 'charlie', 'david', 'eve', 'fred', 'ginny', 'harriet', 'ileana', 'joseph', 'kincaid', 'larry'];
const PLANTS = { V: 'violets', R: 'radishes', C: 'clover', G: 'grass' };

export class Garden {
    constructor(input, students = DEFAULT_STUDENTS) {
        const splitRows = input.split('\n').map((row) => [...row]);
        students.sort().forEach((student, idx) => {
            this[student.toLowerCase()] = [splitRows[0][idx * 2],
                                           splitRows[0][(idx * 2) + 1],
                                           splitRows[1][idx * 2],
                                           splitRows[1][(idx * 2) + 1]].map((plantLetter) => PLANTS[plantLetter]);
        });
    }
}

File: clojure/kindergarten-garden/_iterations/1/src/kindergarten_garden.clj
(ns kindergarten-garden
  (:require [clojure.string :as string]))

(def ^:private plants
  {\G :grass \C :clover \R :radishes \V :violets})

(def ^:private default-students
  ["Alice" "Bob" "Charlie" "David" "Eve" "Fred" "Ginny" "Harriet" "Ileana" "Joseph" "Kincaid" "Larry"])

(defn garden
  ([diagram]
   (garden diagram default-students))
  ([diagram students]
   (let [student-symbols (map #(keyword (string/lower-case %)) (sort students))]
     (->> (string/split-lines diagram)
          (map (partial partition 2))
          (apply map concat)
          (map (partial mapv plants))
          (zipmap student-symbols)))))
